legal-api/src/legal_api/resources/v2/admin/reviews.py

```diff
@@ -35,7 +38,69 @@ def get_reviews():
     return jsonify(reviews), HTTPStatus.OK
 
 
-@bp_admin.route('/reviews/<int:review_id>', methods=['GET', 'OPTIONS'])
+@bp_admin.route('/reviews/<int:review_id>', methods=['POST'])
+@cross_origin(origin='*')
+@jwt.has_one_of_roles([UserRoles.staff])
+def save_review(review_id: int):
+    """Save review.
+
+    Current review status -> Allowable review status
+    AWAITING_REVIEW/RESUBMITTED -> CHANGE_REQUESTED/APPROVED/REJECTED
+    CHANGE_REQUESTED/APPROVED/REJECTED -> No changes allowed
+    """
+    user = User.get_or_create_user_by_jwt(g.jwt_oidc_token_info)
+    if review := Review.find_by_id(review_id):
+        if review.status not in [ReviewStatus.AWAITING_REVIEW, ReviewStatus.RESUBMITTED]:
+            return jsonify({'message': 'No changes allowed.'}), HTTPStatus.BAD_REQUEST
+    else:
+        return jsonify({'message': 'Review not found.'}), HTTPStatus.NOT_FOUND
+
+    json_input = request.get_json()
+    status = json_input.get('status')
+    comment = json_input.get('comment')
+    if not status or not comment:
+        return jsonify({'message': 'Status and Comment are required.'}), HTTPStatus.BAD_REQUEST
```

Comment:
I believe comment is optional when the status is Approved.

Should this be "not status and not comment" for ChangeRequested and Rejected?

Reply:
oh! I thought there will be a default comment from UI

Reply:
This is what the approved payload looks like:

Reply:
* if status or comment is missing

Reply:
> This is what the Approved payload looks like:
> 
> ![image](https://private-user-images.githubusercontent.com/10002889/351483303-0ce3a489-17d9-4b12-bd12-d69b1db6b4f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3NjcwODcsIm5iZiI6MTcyMTc2Njc4NywicGF0aCI6Ii8xMDAwMjg4OS8zNTE0ODMzMDMtMGNlM2E0ODktMTdkOS00YjEyLWJkMTItZDY5YjFkYjZiNGY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDIwMzMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxOTFiZTNmOWMzMDIwNGM1OTMzOWJhZDhkZTJjN2VjMjljZmY2MDUyODlhMDZmMDZlNWQwMzg3NzBhMjM0YmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aYVpqXZpkdXfECmCST7n_j_2GciqAGMcxwf7eXVPP-c)

will change it